app/controllers/api/base_controller.rb

```
@@ -48,17 +48,48 @@ def resource_class
       @resource_class ||= resource_name.classify.constantize
     end
 
-    def resource_scope(controller = controller_name)
+    def resource_scope(controller = controller_name, options = {})
       @resource_scope ||= begin
-        scope = resource_class.scoped
-        if resource_class.respond_to?(:authorized)
-          scope.authorized("#{action_permission}_#{controller}", resource_class)
+        if nested_obj
+          resource_scope_nested(controller, options)
         else
-          scope
+          resource_scope_not_nested(controller, options)
         end
       end
     end
 
+    def resource_scope_not_nested(controller = controller_name, options = {})
+      scope = resource_class.scoped
+      options[:permission] ||= "#{action_permission}_#{controller}"
+      if resource_class.respond_to?(:authorized)
```

this could lead to silently ignored :permission option, it would be better to raise when :permission is supplied but not supported by the resource_class